pkg/datasources/secrets_acceptance_test.go

```diff
@@ -0,0 +1,305 @@
+package datasources_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert"
+	accConfig "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/config/model"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+const (
+	secretWithClientCredentials      = "snowflake_secret_with_client_credentials"
+	secretWithAuthorizationCodeGrant = "snowflake_secret_with_authorization_code_grant"
+	secretWithBasicAuthentication    = "snowflake_secret_with_basic_authentication"
+	secretWithGenericString          = "snowflake_secret_with_generic_string"
+)
+
+func TestAcc_Secrets_MultipleTypes(t *testing.T) {
+	prefix := random.AlphaN(4)
+	idOne := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "A")
+	idTwo := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "B")
+	idThree := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "C")
+	idFour := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix + "D")
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
```

Comment:
if you want to run such statements before PreConfig, then you should use EnableAcceptance and acc.Precheck at the start of the test functions (check e.g. TestAcc_SecurityIntegrations_OauthForCustomClients).

The reasons: currently the setup for acceptance tests is run in acc.Precheck, so usage of `acc.TestClient()` may result in strange behavior. The EnableAcceptance check prevents running it when running integration tests (for now we do go test with filters, and terraform testing framework checks for TF_ACC late)

Reply:
Is `acc.TestAccPreCheck(t)` sufficient or is there a need for `testenvs.GetOrSkipTest()` just like in the `TestAcc_SecurityIntegrations_OauthForCustomClients` as well?

Reply:
GetOrSkipTest is needed too (that's what I meant by saying use EnableAcceptance, sorry)